Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs

```diff
@@ -36,39 +32,41 @@ public void ExitDisposals(EntityUid uid, DisposalHolderComponent? holder = null,
             // Check for a disposal unit to throw them into and then eject them from it.
             // *This ejection also makes the target not collide with the unit.*
             // *This is on purpose.*
-            var grid = _mapManager.GetGrid(holderTransform.GridID);
-            var gridTileContents = grid.GetLocal(holderTransform.Coordinates);
+
             DisposalUnitComponent? duc = null;
-            foreach (var contentUid in gridTileContents)
+            if (_mapManager.TryGetGrid(holderTransform.GridID, out var grid))
             {
-                if (EntityManager.TryGetComponent(contentUid, out duc))
-                    break;
+                foreach (var contentUid in grid.GetLocal(holderTransform.Coordinates))
+                {
+                    if (EntityManager.TryGetComponent(contentUid, out duc))
+                        break;
+                }
             }
 
             foreach (var entity in holder.Container.ContainedEntities.ToArray())
             {
-                if (HasComp<BeingDisposedComponent>(entity))
-                    RemComp <BeingDisposedComponent>(entity);
+                RemComp<BeingDisposedComponent>(entity);
 
                 if (EntityManager.TryGetComponent(entity, out IPhysBody? physics))
                 {
                     physics.CanCollide = true;
                 }
 
-                holder.Container.ForceRemove(entity);
+                holder.Container.ForceRemove(entity, EntityManager);
 
-                if (EntityManager.GetComponent<TransformComponent>(entity).Parent == holderTransform)
+                var xform = EntityManager.GetComponent<TransformComponent>(entity);
+                if (xform.Parent != holderTransform)
```

Comment:
Parent does a GetComponent internally so you should always use ParentUid instead where possible.

Should also use the Transform() proxy.

Reply:
ah yeah didn't spot that. will fix